src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -191,13 +191,29 @@ public RuntimeConfig(
         this.Entities = Entities;
         this.DefaultDataSourceName = Guid.NewGuid().ToString();
 
+        if (this.DataSource is null)
+        {
+            throw new DataApiBuilderException(
+                message: "DataSource is a mandatory property in DAB Config",
+                statusCode: HttpStatusCode.UnprocessableEntity,
```

Comment:
I think that the message needs to be a bit more specific to show that the datasource is mandatory in all config files
```suggestion
                message: "DataSource is a mandatory property in all DAB configuration files",
```

Reply:
Since we are validating config file passed in the parameter `c {configfilename}`. This shouldnt be needed

Reply:
I think it will be useful if we can identify the configuration file name that has the missing data-source  when its the multiple config scenario.

Reply:
Even in case of multiple data sources, we only validate the parent or top level data source file so I dont think we would need to mention the name.

Reply:
Is this behavior that `validate` operates only on the top level data source known in our documentation? I thought it would do the validation recursively..

Reply:
No thats not the scenario currently. Already added an issue to track this  [Enhance Validate](https://github.com/Azure/data-api-builder/issues/2493)